packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/Tasks.spec.js

```diff
@@ -1,133 +1,170 @@
-  const annotation = classifications.addAnnotation(singleChoiceTask)
-  const classification = classifications.active
-  const { addAnnotation } = classifications
-  singleChoiceTask.setAnnotation(annotation)
-
-  it('should render without crashing', function () {
-    const wrapper = shallow(<Tasks />)
-    expect(wrapper).to.be.ok()
-  })
+  let addAnnotation
+  let classification
+  let step
+  let tasks
+  let TaskComponent
 
-  it('should render null on initialization', function () {
-    const wrapper = shallow(<Tasks />)
-    expect(wrapper.type()).to.be.null()
-  })
+  const taskTypes = Object.keys(taskRegistry.register)
 
-  it('should render a loading UI when the workflow loading', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.loading} />)
-    expect(wrapper.contains('Loading')).to.be.true()
+  before(function () {
+    sinon.stub(console, 'error')
   })
 
-  it('should render an error message when there is a loading error', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.error} />)
-    expect(wrapper.contains('Something went wrong')).to.be.true()
+  after(function () {
+    console.error.restore()
   })
 
-  it('should render null if the workflow is load but has no tasks', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.success} ready />)
-    expect(wrapper.type()).to.be.null()
-  })
+  taskTypes.forEach(function (taskType) {
+    before(function () {
+      const task = taskRegistry.get(taskType)
+      TaskComponent = observer(task.TaskComponent)
+      const taskSnapshot = {
+        instruction: `${taskType} instructions`,
+        question: `${taskType} question`,
+        taskKey: 'init',
+        type: taskType
+      }
+      const workflowSnapshot = WorkflowFactory.build({
+        id: 'tasksWorkflow',
+        display_name: 'A test workflow',
+        tasks: {
+          init: taskSnapshot
+        },
+        version: '0.0'
+      })
+      const subjectSnapshot = SubjectFactory.build({
+        id: 'subject',
+        metadata: {}
+      })
+      const projectSnapshot = ProjectFactory.build({
+        id: 'project'
+      })
+      const rootStore = RootStore.create({
+        projects: {
+          active: projectSnapshot.id,
+          resources: {
+            [projectSnapshot.id]: projectSnapshot
+          }
+        },
+        subjects: {
+          active: subjectSnapshot.id,
+          resources: {
+            [subjectSnapshot.id]: subjectSnapshot
+          }
+        },
+        workflows: {
+          active: workflowSnapshot.id,
+          resources: {
+            [workflowSnapshot.id]: workflowSnapshot
+          }
+        }
+      }, {
+        client: {
+          panoptes: {
+            get: sinon.stub().callsFake(() => Promise.resolve({ body: {}}))
+          }
+        }
+      })
+      classification = rootStore.classifications.active
+      addAnnotation = rootStore.classifications.addAnnotation
```

Comment:
If we try to use destructuring here
```js
({ addAnnotation } = rootStore.classifications)
```
then the interpreter, without any semicolons to guide it, will try to execute `classifications.active` as a function and throw an error.
```js
classification = rootStore.classifications.active({ addAnnotation } = rootStore.classifications)
```

Reply:
I don't really have a strong opinion about linter rules other than consistency and also not spending any time on deciding what those rules should be... 

Reading [the docs from standard](https://standardjs.com/rules.html#semicolons), I think they'd suggest to do something like:
`;({ addAnnotation } = rootStore.classifications)`

But as I said, I don't actually care what the rule is all that much, just that we stay consistent. There's a variant of standard that uses semi-colons if you feel strongly about it: https://github.com/standard/semistandard

Reply:
No strong opinions from me except to be mindful that the interpreter doesn’t always insert semicolons where you’d expect. When writing/reviewing, be wary of declarations that begin with `(` or `[`.